DevOps/DynamoDBAutomation/dynamoDBbuilder.py

```diff
@@ -0,0 +1,49 @@
+# reads the createDynamoDB.json file and creates a stack
+import json
+import boto3
+
+
+# wrap the operation in a function
+
+def create_dynamodb_table(tableName, keyName, cfTemplate):
+    """Given a table name, key name and CFT template, it'll create a DynamoDB table.
+
+    Keyword arguments:
+    tableName -- the DyanmoDB table name
+    keyName -- the partion key name
+    cfTemplate -- the file path to the cloud formation template json
+    """
+    cfnclient = boto3.client('cloudformation')
+
+    # open the template file
+    with open(cfTemplate) as fd:
+        template = json.load(fd)
+
+    # update the template with the table name and hash key name
+    template["Resources"]["DynamoDBTable"]["Properties"]["TableName"] = tableName
+    template["Resources"]["DynamoDBTable"]["Properties"]["AttributeDefinitions"][0]["AttributeName"] = keyName
+    template["Resources"]["DynamoDBTable"]["Properties"]["KeySchema"][0]["AttributeName"] = keyName
+
+
+# create the stack
+    cfnclient.create_stack(StackName=tableName,
+                           TemplateBody=json.dumps(template)
+                           )
+
+# wait for the stack to complete
+    waiter = cfnclient.get_waiter('stack_create_complete')
+    waiter.wait(StackName=tableName)
+
+# get the stack output
+    stack = cfnclient.describe_stacks(StackName=tableName)
+    table_status = stack['Stacks'][0]['StackStatus']
+
+# return the table status
+    return table_status
+
+
+# test the function
```

Comment:
I would un-comment these

Reply:
Those are deliberately done. Checkout the "To Test"> 'steps'

Reply:
Instead of leaving them here and saying...

> "un comment line 46-49 from dynamoDBbuilder.py file"

... just add them to the testing steps, and tell the tester to copy/paste them into the code to test.

This way you won't be committing unneeded code comments to the repo.

Reply:
Nice idea, Thanks!